### ISSUE TITLE
Add support for Kotlin co-routines

### ISSUE THREAD

**Issue by juergenzimmermann** [spring-projects/spring-hateoas#1213]
It would be nice, if `SimpleReactiveRepresentationModelAssembler.toCollectionModel()` would support Kotlin's `Flow` in addition to Reactor's `Flux`.

I tried the following extension function which works fine:
```Kotlin
suspend fun <T: Any> SimpleReactiveRepresentationModelAssembler<T>.toCollectionModel(
    entities: Flow<T>,
    exchange: ServerWebExchange
): CollectionModel<EntityModel<T>> = toCollectionModel(entities.asFlux(), exchange).awaitFirst()
```

**Comment by gregturn** [comment]
#1258 widens this Kotlin support.

**Comment by gregturn** [comment]
Resolved via b46082b1.

Thanks @juergenzimmermann!
